beacon_node/lighthouse_network/src/rpc/protocol.rs

```diff
@@ -328,13 +328,13 @@ impl ProtocolId {
     /// Returns `true` if the given `ProtocolId` should expect `context_bytes` in the
     /// beginning of the stream, else returns `false`.
     pub fn has_context_bytes(&self) -> bool {
-        if self.version == Version::V2 {
-            match self.message_name {
-                Protocol::BlocksByRange | Protocol::BlocksByRoot => return true,
-                _ => return false,
+        match self.message_name {
+            Protocol::BlocksByRange | Protocol::BlocksByRoot => {
+                !matches!(self.version, Version::V1)
```

Comment:
Actually, it might be better if we explicitly match `Version::V2` instead of excluding V1. If there's a V3 in the future, can't say for sure if that would have a similar mechanism for context bytes.

Reply:
Added this and similar for the light client endpoint, but the fact that we're now going to handle a light client protocol + version combination that doesn't exist let me to thinking about this more broadly: https://github.com/sigp/lighthouse/issues/3980